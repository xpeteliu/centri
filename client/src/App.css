.mainContent {
  margin: 20px;
}

body {
  background-color: #E4F0D1;
}

.btn-primary, .btn-primary:hover, .btn-primary:active, .btn-primary:visited {
  background-color: #96BB88 !important;
  border-color: #96BB88 !important;
  color: #000000 !important;
}

.btn-secondary, .btn-secondary:hover, .btn-secondary:active, .btn-secondary:visited {
  background-color: #C1D9B8 !important;
  border-color: #C1D9B8 !important;
  color: #000000 !important;
}

.headerBar {
  background-color: #C1D9B8 !important;
}

.groupListItem {
  background-color: #FFFFFF;
  height: 80px;
  padding: 20px;
}

/* .App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-float infinite 3s ease-in-out;
  }
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.App-link {
  color: rgb(112, 76, 182);
}

@keyframes App-logo-float {
  0% {
    transform: translateY(0);
  }
  50% {
    transform: translateY(10px);
  }
  100% {
    transform: translateY(0px);
  }
} */
