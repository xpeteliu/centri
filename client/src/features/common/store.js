import { configureStore } from '@reduxjs/toolkit';
import modalReducer from './MessageModal/modalSlice';

const store = configureStore({
  reducer: {
    modal: modalReducer,
  },
});

export default store;
