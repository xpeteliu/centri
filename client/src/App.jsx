import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom';
import HomePage from './components/HomePage/HomePage';
import LoginPage from './components/LoginPage/LoginPage';
import './App.css';

const Home = () => (
  <HomePage />
);

const Login = () => (
  <LoginPage />
);

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <Route exact path="/" component={Home} />
          <Route path="/login" component={Login} />
        </div>
      </Router>
    );
  }
}

export default App;
