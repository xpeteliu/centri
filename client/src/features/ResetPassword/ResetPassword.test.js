import React from 'react';
import {
  act, render, screen, waitFor,
} from '@testing-library/react';
import { Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { setupServer } from 'msw/node';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { Provider } from 'react-redux';
import App from '../../App';
import store from '../common/store';

describe('Registration page', () => {
  const mockServer = setupServer(
    rest.post('/api/user/filter/paginate',
      (req, res, ctx) => (
        res(ctx.json(req.body.filter.email === 'testName@email.com' ? [{
          _id: 'abc123abc123',
          username: 'testName',
          email: 'testName@email.com',
        }] : []))
      )),
    rest.put('/api/user/:id', (req, res, ctx) => res(ctx.json({
      _id: 'abc123abc123',
      username: 'testName',
      email: 'testName@email.com',
    }))),
  );

  beforeAll(() => {
    mockServer.listen();
  });

  afterAll(() => {
    mockServer.close();
  });

  test('should match snapshot', async () => {
    let container;
    act(() => {
      ({ container } = render(
        <Provider store={store}>
          <Router history={createMemoryHistory({ initialEntries: ['/resetPassword'] })}>
            <App />
          </Router>
        </Provider>,
      ));
    });
    expect(container)
      .toMatchSnapshot();
  });

  test('should properly handle valid reset request', async () => {
    const history = createMemoryHistory({ initialEntries: ['/resetPassword'] });
    act(() => {
      render(
        <Provider store={store}>
          <Router history={history}>
            <App />
          </Router>
        </Provider>,
      );
    });
    const submitBtn = await screen.findByRole('button', { name: /Reset/i });
    const usernameInput = await screen.findByLabelText('Username');
    const emailInput = await screen.findByLabelText('Email');
    const passwordInput = await screen.findByLabelText('New Password');
    act(() => {
      userEvent.type(usernameInput, 'testName');
      userEvent.type(emailInput, 'testName@email.com');
      userEvent.type(passwordInput, 'testPwd');
      userEvent.click(submitBtn);
    });
    await waitFor(() => {
      expect(history.location.pathname)
        .toEqual('/login');
    });
  });

  test('should reject wrong email info', async () => {
    const history = createMemoryHistory({ initialEntries: ['/resetPassword'] });
    act(() => {
      render(
        <Provider store={store}>
          <Router history={history}>
            <App />
          </Router>
        </Provider>,
      );
    });
    const submitBtn = await screen.findByRole('button', { name: /Reset/i });
    const usernameInput = await screen.findByLabelText('Username');
    const emailInput = await screen.findByLabelText('Email');
    const passwordInput = await screen.findByLabelText('New Password');
    act(() => {
      userEvent.type(usernameInput, 'testName');
      userEvent.type(emailInput, 'wrong@email.com');
      userEvent.type(passwordInput, 'testPwd');
      userEvent.click(submitBtn);
    });
    expect(await screen.findByText(/Unable to Reset Password/i))
      .toBeInTheDocument();
  });
});
