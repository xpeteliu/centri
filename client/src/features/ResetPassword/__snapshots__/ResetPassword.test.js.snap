// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration page should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="headerBar navbar navbar-expand-md navbar-light bg-light"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
          role="presentation"
        >
          Team20
        </span>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
        >
          <div
            class="me-auto navbar-nav"
          />
          <div
            class="hstack gap-2"
          >
            <span
              class="navbar-text"
            />
          </div>
        </div>
      </div>
    </nav>
    <div
      class="h-100 container"
    >
      <div
        class="h-100 row"
      >
        <div
          class="my-auto col"
        >
          <div
            class="mx-auto card"
            style="width: 30rem;"
          >
            <div
              class="card-header"
            >
              <h2>
                Reset Password
              </h2>
            </div>
            <div
              class="card-body"
            >
              <form
                class="container"
                id="formLogin"
                style="width: 25rem;"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 text-start"
                    >
                      <label
                        class="ms-0 form-label"
                        for="inputResetUsername"
                      >
                        Username
                      </label>
                      <input
                        autocomplete="username"
                        class="form-control"
                        id="inputResetUsername"
                        placeholder="Enter username here"
                        required=""
                        type="text"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 text-start"
                    >
                      <label
                        class="form-label"
                        for="inputResetEmail"
                      >
                        Email
                      </label>
                      <input
                        autocomplete="email"
                        class="form-control"
                        id="inputResetEmail"
                        placeholder="Enter email here"
                        required=""
                        type="email"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 text-start"
                    >
                      <label
                        class="form-label"
                        for="inputResetPassword"
                      >
                        New Password
                      </label>
                      <input
                        autocomplete="new-password"
                        class="form-control"
                        id="inputResetPassword"
                        pattern="^(?=.*[A-Za-z])(?=.*\\\\d)[A-Za-z\\\\d]{8,}$"
                        placeholder="Enter password here"
                        required=""
                        title="A password should contain >=8 characters and include both letters and numbers"
                        type="password"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                  style="margin-top: 1rem;"
                >
                  <div
                    class="col"
                  >
                    <button
                      class="btn btn-primary"
                      form="formLogin"
                      type="submit"
                    >
                      Reset
                    </button>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="container card-footer"
            >
              <div
                class="text-center row"
              >
                <div
                  class="col"
                >
                  <a
                    class="card-link"
                    href="/login"
                  >
                    Already have an account?
                  </a>
                </div>
                <div
                  class="col"
                >
                  <a
                    class="card-link"
                    href="/"
                  >
                    Back to Home Page
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
